src/main/java/com/example/UserService.java

```diff
@@ -0,0 +1,29 @@
+package com.example;
+
+import java.util.List;
+
+public class userService {
+    private List<String> users;
+
+    // Constructor injection
+    public UserService(List<String> users) {
+        this.users = users;
+    }
+
+    // Rest of the code...
+
+    public boolean isUserExists(String userName) {
+        boolean f = false;
+
+        for (int index = 0; index < users.size(); index++) {
```

Comment:
have you considered using [contains()](https://www.geeksforgeeks.org/list-contains-method-in-java-with-examples/) or [stream().anyMatch(x -> x.equals(userName)](https://www.geeksforgeeks.org/stream-anymatch-java-examples/) instead? these may be a bit more efficient!